python_modules/dagster/dagster/_config/errors.py

```diff
@@ -23,80 +24,44 @@ class DagsterEvaluationErrorReason(Enum):
     FIELD_ALIAS_COLLISION = "FIELD_ALIAS_COLLISION"
 
 
-class FieldsNotDefinedErrorData(
-    NamedTuple("_FieldsNotDefinedErrorData", [("field_names", Sequence[str])])
-):
-    def __new__(cls, field_names: Sequence[str]):
-        return super(FieldsNotDefinedErrorData, cls).__new__(
-            cls, check.sequence_param(field_names, "field_names", of_type=str)
-        )
+@dagster_model
+class FieldsNotDefinedErrorData:
+    field_names: Sequence[str]
 
 
-class FieldAliasCollisionErrorData(NamedTuple):
+@dagster_model
+class FieldAliasCollisionErrorData:
     field_name: str
     aliased_field_name: str
 
 
-class FieldNotDefinedErrorData(NamedTuple("_FieldNotDefinedErrorData", [("field_name", str)])):
-    def __new__(cls, field_name: str):
-        return super(FieldNotDefinedErrorData, cls).__new__(
-            cls, check.str_param(field_name, "field_name")
-        )
+@dagster_model
+class FieldNotDefinedErrorData:
+    field_name: str
 
 
-class MissingFieldErrorData(
-    NamedTuple("_MissingFieldErrorData", [("field_name", str), ("field_snap", ConfigFieldSnap)])
-):
-    def __new__(cls, field_name, field_snap):
-        return super(MissingFieldErrorData, cls).__new__(
-            cls,
-            check.str_param(field_name, "field_name"),
-            check.inst_param(field_snap, "field_snap", ConfigFieldSnap),
-        )
+@dagster_model
+class MissingFieldErrorData:
+    field_name: str
+    field_snap: ConfigFieldSnap
 
 
-class MissingFieldsErrorData(
-    NamedTuple(
-        "_MissingFieldErrorData",
-        [("field_names", Sequence[str]), ("field_snaps", Sequence[ConfigFieldSnap])],
-    )
-):
-    def __new__(cls, field_names, field_snaps):
-        return super(MissingFieldsErrorData, cls).__new__(
-            cls,
-            check.list_param(field_names, "field_names", of_type=str),
-            check.list_param(field_snaps, "field_snaps", of_type=ConfigFieldSnap),
-        )
-
-
-class RuntimeMismatchErrorData(
-    NamedTuple(
-        "_RuntimeMismatchErrorData", [("config_type_snap", ConfigTypeSnap), ("value_rep", str)]
-    )
-):
-    def __new__(cls, config_type_snap, value_rep):
-        check.inst_param(config_type_snap, "config_type", ConfigTypeSnap)
-        return super(RuntimeMismatchErrorData, cls).__new__(
-            cls,
-            config_type_snap,
-            check.str_param(value_rep, "value_rep"),
-        )
-
-
-class SelectorTypeErrorData(
-    NamedTuple(
-        "_SelectorTypeErrorData",
-        [("config_type_snap", ConfigTypeSnap), ("incoming_fields", Sequence[str])],
-    )
-):
-    def __new__(cls, config_type_snap, incoming_fields):
-        check.inst_param(config_type_snap, "config_type_snap", ConfigTypeSnap)
-        check.param_invariant(config_type_snap.kind == ConfigTypeKind.SELECTOR, "config_type")
```

Comment:
knee jerk choose to just drop this, could bring it back

Reply:
would rather make this a change that changes no behavior